src/DataInitializer.java

```diff
@@ -0,0 +1,43 @@
+import java.io.File;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class DataInitializer {
+    private static final DataInitializer INSTANCE;
+
+    static {
+        INSTANCE = new DataInitializer();
+    }
```

Comment:
static block not needed.
```suggesstion
    private static final DataInitializer INSTANCE = new DataInitializer();
```

Reply:
according to my searches using the static block is not mentioned as a bad practice how ever for making instance is better to use in line initialization . 
Thank You Mr.Abedi